modules/plugin/jdbc/jdbc-sqlserver/src/main/java/org/geotools/data/sqlserver/SQLServerDialect.java

```diff
@@ -165,6 +166,9 @@ public void registerSqlTypeNameToClassMappings(Map<String, Class<?>> mappings) {
         mappings.put("uniqueidentifier", UUID.class);
         mappings.put("time", Time.class);
         mappings.put("date", Date.class);
+        mappings.put("datetime", Timestamp.class);
```

Comment:
We have an existing testcase that checks a `datetime` column in SQLServerDateOnlineTest ⬅️ JDBCDateOnlineTest

https://github.com/geotools/geotools/blob/01e27991725d1394b683439a774c4827493d6f95/modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCDateOnlineTest.java#L24-L29bc/src/test/java/org/geotools/jdbc/JDBCDateOnlineTest.java#L24-L29

but no tests for datetime2 or datetimeoffset